### ISSUE TITLE
Parsing Errors from Example

### ISSUE THREAD

**Issue by Reubend** [pytorch/kineto#269]
I have been following the [tutorial for using the PyTorch profiler](https://pytorch.org/tutorials/intermediate/tensorboard_profiler_tutorial.html), and I've already logged the data to a local directory. I made some minor modifications to the code (exact code below in a comment) but nothing which should affect the profiler's accuracy. 

When I run the TB plugin for this directory, I get hundreds of error messages like these inside of the terminal:
```
E0601 09:46:01.455774 140059630905088 module_parser.py:136] Error in input data: ranges on the same thread should not intersect!Father:(aten::copy_,1622565333177893,1622565333177968) Child:(aten::copy_,1622565333177901,1622565333177979)
E0601 09:46:01.455836 140059630905088 module_parser.py:136] Error in input data: ranges on the same thread should not intersect!Father:(aten::to,1622565333178010,1622565333178120) Child:(aten::to,1622565333178022,1622565333178127)
```

These errors seem to be coming from [this line of code](https://github.com/pytorch/kineto/blob/8745f178861c34ab5e8f448dc6eb03f6e9700641/tb_plugin/torch_tb_profiler/profiler/module_parser.py#L136). I can still load TB at the expected URL though.

**Expected behavior:** 
Tutorial output loads without any issues
**Actual behavior:**
Errors appear from the tutorial output

**Comment by Reubend** [comment]
Additional info to help with debugging: I changed the example code in two ways. I disabled GPU training, and reduced the number of training steps. Here's the exact code I ran:
```python
import torch
import torch.nn
import torch.optim
import torch.profiler
import torch.utils.data
import torchvision.datasets
import torchvision.models
import torchvision.transforms as T

transform = T.Compose(
    [T.Resize(224),
     T.ToTensor(),
     T.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))])
train_set = torchvision.datasets.CIFAR10(root='./data', train=True, download=True, transform=transform)
train_loader = torch.utils.data.DataLoader(train_set, batch_size=32, shuffle=True)

model = torchvision.models.resnet18(pretrained=True)
criterion = torch.nn.CrossEntropyLoss()
optimizer = torch.optim.SGD(model.parameters(), lr=0.001, momentum=0.9)
model.train()

def train(data):
    inputs, labels = data[0], data[1]
    outputs = model(inputs)
    loss = criterion(outputs, labels)
    optimizer.zero_grad()
    loss.backward()
    optimizer.step()

with torch.profiler.profile(
        schedule=torch.profiler.schedule(wait=1, warmup=1, active=5),
        on_trace_ready=torch.profiler.tensorboard_trace_handler('./log/resnet18'),
        record_shapes=True
) as prof:
    for step, batch_data in enumerate(train_loader):
        if step >= 2:
            break
        train(batch_data)
        prof.step()  # Need call this at the end of each step to notify profiler of steps' boundary.
```

**Comment by gaoteng-git** [comment]
> Additional info to help with debugging: I changed the example code in two ways. I disabled GPU training, and reduced the number of training steps. Here's the exact code I ran:
> 
> ```python
> import torch
> import torch.nn
> import torch.optim
> import torch.profiler
> import torch.utils.data
> import torchvision.datasets
> import torchvision.models
> import torchvision.transforms as T
> 
> transform = T.Compose(
>     [T.Resize(224),
>      T.ToTensor(),
>      T.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))])
> train_set = torchvision.datasets.CIFAR10(root='./data', train=True, download=True, transform=transform)
> train_loader = torch.utils.data.DataLoader(train_set, batch_size=32, shuffle=True)
> 
> model = torchvision.models.resnet18(pretrained=True)
> criterion = torch.nn.CrossEntropyLoss()
> optimizer = torch.optim.SGD(model.parameters(), lr=0.001, momentum=0.9)
> model.train()
> 
> def train(data):
>     inputs, labels = data[0], data[1]
>     outputs = model(inputs)
>     loss = criterion(outputs, labels)
>     optimizer.zero_grad()
>     loss.backward()
>     optimizer.step()
> 
> with torch.profiler.profile(
>         schedule=torch.profiler.schedule(wait=1, warmup=1, active=5),
>         on_trace_ready=torch.profiler.tensorboard_trace_handler('./log/resnet18'),
>         record_shapes=True
> ) as prof:
>     for step, batch_data in enumerate(train_loader):
>         if step >= 2:
>             break
>         train(batch_data)
>         prof.step()  # Need call this at the end of each step to notify profiler of steps' boundary.
> ```

I haven't reproduced this problem running your code. Would you please share your json file to me? My email is tegao@microsoft.com
And also it's better to share your configuration which is got by "python -m torch.utils.collect_env".

**Comment by gaoteng-git** [comment]
> I have been following the [tutorial for using the PyTorch profiler](https://pytorch.org/tutorials/intermediate/tensorboard_profiler_tutorial.html), and I've already logged the data to a local directory. I made some minor modifications to the code (exact code below in a comment) but nothing which should affect the profiler's accuracy.
> 
> When I run the TB plugin for this directory, I get hundreds of error messages like these inside of the terminal:
> 
> ```
> E0601 09:46:01.455774 140059630905088 module_parser.py:136] Error in input data: ranges on the same thread should not intersect!Father:(aten::copy_,1622565333177893,1622565333177968) Child:(aten::copy_,1622565333177901,1622565333177979)
> E0601 09:46:01.455836 140059630905088 module_parser.py:136] Error in input data: ranges on the same thread should not intersect!Father:(aten::to,1622565333178010,1622565333178120) Child:(aten::to,1622565333178022,1622565333178127)
> ```
> 
> These errors seem to be coming from [this line of code](https://github.com/pytorch/kineto/blob/8745f178861c34ab5e8f448dc6eb03f6e9700641/tb_plugin/torch_tb_profiler/profiler/module_parser.py#L136). I can still load TB at the expected URL though.
> 
> **Expected behavior:**
> Tutorial output loads without any issues
> **Actual behavior:**
> Errors appear from the tutorial output

I haven't reproduced this problem running your code. Would you please share your json file to me? My email is tegao@microsoft.com
And also it's better to share your configuration which is got by "python -m torch.utils.collect_env".

**Comment by Reubend** [comment]
Thanks @gaoteng-git I've emailed you the trace files which were generated. 

As for the environment info, I don't think this will be too helpful as I'm running this inside of a fairly locked down environment:
```
PyTorch version: 1.9.0a0
Is debug build: False
CUDA used to build PyTorch: 11.0.0
ROCM used to build PyTorch: N/A

OS: CentOS Stream 8 (x86_64)
GCC version: (GCC) 8.4.1 20200928 (Red Hat 8.4.1-1)
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.30

Python version: 3.8 (64-bit runtime)
Python platform: Linux-5.2.9
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip] Could not collect
[conda] Could not collect
```

But I'm happy to get the additional info manually if you need. I'm using the 0.1.0 version of `torch-tb-profiler`.

**Comment by gaoteng-git** [comment]
@Reubend Thanks! I checked your json file, and these warnings are caused by overlapped operators, which is not expected behavior.
@gdankel It seems kineto or pytorch profiler's problem. Would you please have a look?
